mutation/position_in_partition.hh

```diff
@@ -361,13 +361,13 @@ public:
         }
 
     position_in_partition& operator=(position_in_partition_view view) {
-        _type = view._type;
-        _bound_weight = view._bound_weight;
         if (view._ck) {
             _ck = *view._ck;
         } else {
             _ck.reset();
         }
+        _type = view._type;
+        _bound_weight = view._bound_weight;
```

Comment:
We should document that this method is supposed to provide strong exception guarantees, so that this impl doesn't get broken accidentally later.

Reply:
Done.